docs/walkthroughs/upgrading-v2-to-v3.md

```diff
@@ -0,0 +1,301 @@
+                    stat_prefix: ingress_http
+                    route_config:
+                      name: route
+                      virtual_hosts:
+                        - name: any
+                          domains: ["*"]
+                          routes:
+                            - {"match":{"prefix":"/"},"direct_response":{"status":200}}
+                    http_filters:
+                      - name: envoy.filters.http.router
+EOF
+```
+
+Immediately after applying the v3 config, a rollout will occur to update the EnvoyDeployment to use v3:
+
+```bash
+▶ kubectl get pods
+NAME                                             READY   STATUS    RESTARTS   AGE
+marin3r-discoveryservice-7b4c6d4c7c-fftg2        1/1     Running   0          64m
+marin3r-envoydeployment-envoy-7494674d9d-7nx9h   0/1     Running   0          28s
+marin3r-envoydeployment-envoy-7fd98988c-9m9n2    1/1     Running   0          53m
+```
+
+After the Deployment rollout is complete, the new Envoy pods will be using v3.
+
+```bash
+▶ kubectl get pods
+NAME                                             READY   STATUS    RESTARTS   AGE
+marin3r-discoveryservice-7b4c6d4c7c-fftg2        1/1     Running   0          65m
+marin3r-envoydeployment-envoy-7494674d9d-7nx9h   1/1     Running   0          77s
+
+▶ kubectl get envoyconfig
+NAME    NODE ID   ENVOY API   DESIRED VERSION   PUBLISHED VERSION   CACHE STATE
+envoy   envoy     v3          76795866f         76795866f           InSync
+```
+
+It's important to note that during the rollout process, both the v2 and the v3 versions of the config are being served by the discovery service. This is designed this way because during the rollout both v2 and v3 clients will be requesting configurations to the discovery service. In fact, the v2 EnvoyConfigRevisions (the internal operator resource that the EnvoyConfig controller to keep track of config versioning) will be never deleted by the operator, as we can see using the following command:
+
+```bash
+▶ kubectl get envoyconfigrevisions
+NAME                 NODE ID   ENVOY API   VERSION     PUBLISHED   CREATED AT             LAST PUBLISHED AT      TAINTED
+envoy-696bd4bdc      envoy     v2          696bd4bdc   true        2021-07-07T09:39:42Z   2021-07-07T09:39:42Z
+envoy-v3-76795866f   envoy     v3          76795866f   true        2021-07-07T10:33:17Z   2021-07-07T10:33:17Z
+```
+
+### Cleanup
+
+Execute the following commands to delete the resources created in this walkthough:
+
+```bash
+kubectl delete envoydeployment envoy
+kubectl delete envoyconfig envoy
+```
+
+## **Procedure for Envoy sidecars**
+
+The process to upgrade Envoy sidecars from v2 to v3 involves an extra step due to the fact that when using sidecars the operator has no control over the lifecycle of the pods themselves, it just injects sidecar containers whenever a new Pod is created.
```

Comment:
```suggestion
The process to upgrade Envoy sidecars from v2 to v3 involves an extra step since when using sidecars the operator has no control over the lifecycle of the pods themselves, it just injects sidecar containers whenever a new Pod is created.
```